### PR TITLE
Chore/consultation data validation

### PR DESCRIPTION
**Pull request by Acetoshi**
added data validation and tests in the consultation resolver

### CODE REVIEW

File: core_api/src/modules/consultation/consultation.test.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+export const CREATE_CONSULTATION = gql`
+  mutation CreateConsultation($consultationDetails: CreateConsultationInput!) {
+    createConsultation(consultationDetails: $consultationDetails) {
+      id
+    }
+  }
+`;
+
+// variableValues: { consultationDetails: {
```
Comment (teasmade) [inline core_api/src/modules/consultation/consultation.test.ts]: code mort?

Comment (julien44830) [inline core_api/src/modules/consultation/consultation.test.ts]: du code mort ?
```
// variableValues: { consultationDetails: {
//   description: details?.description,
//   end: consultationDateTime?.end.toISOString() as string,
//   start: consultationDateTime?.start.toISOString() as string,
//   doctorId: doctorId,
//   patientId: patientId,
//   subjectLabel: details.subject
// } }
```

File: core_api/src/modules/consultation/consultation.test.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+          patientId: '4565464sdfsd-sdfsdf74'
+        }
+      }
+    });
+
+    expect(result.errors).toEqual(expect.any(Array));
+    expect(result.data).toBeNull();
+  });
+
+  // Edge cases
```
Comment (julien44830) [inline core_api/src/modules/consultation/consultation.test.ts]: what ? ` // Edge cases`

**Review by teasmade — DISMISSED** [review]

**Review by julien44830 — DISMISSED** [review]

**Review by julien44830 — APPROVED** [review]
